pkg/daemon/daemon.go

```diff
@@ -277,11 +303,20 @@ func (d *daemon) updatePodNetworkAnnotation(pi *podNetworkInfo, removedList *[]n
 
 	pi.pod.Annotations[v1.NetworkAttachmentAnnot] = string(netAnnotations)
 
-	if err := d.kubeClient.SetAnnotationsOnPod(pi.pod, pi.pod.Annotations); err != nil {
-		if kerrors.IsNotFound(err) {
-			return fmt.Errorf("failed to update pod annotations with err: %v", err)
+	// Try to set pod's annotations in backoff loop
+	if err = wait.ExponentialBackoff(backoffValues, func() (bool, error) {
+		if err = d.kubeClient.SetAnnotationsOnPod(pi.pod, pi.pod.Annotations); err != nil {
+			if kerrors.IsNotFound(err) {
+				return false, err
+			}
+			log.Warn().Msgf("failed to update pod annotations with err: %v", err)
+			return false, nil
 		}
 
+		return true, nil
+	}); err != nil {
+		log.Error().Msgf("failed to update pod annotations")
```

Comment:
should we also add the last error to the returned error for completeness ?

Reply:
same answer as previous